From c82b6f987e97f7d0df4c826c6e58aae43e638d85 Mon Sep 17 00:00:00 2001
From: Adrien Ricciardi <aricciardi@baylibre.com>
Date: Fri, 6 Oct 2023 10:36:38 +0200
Subject: [PATCH 24/38] net: can: Removed the now useless protocol version
 field from a CAN controller configuration.

The CAN protocol version can now be retrieved from the baud rate configuration
in a way that is more compliant with AUTOSAR.

Signed-off-by: Adrien Ricciardi <aricciardi@baylibre.com>
---
 net/can/Can_GeneralTypes.h | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/net/can/Can_GeneralTypes.h b/net/can/Can_GeneralTypes.h
index 9390a735..0885ef33 100644
--- a/net/can/Can_GeneralTypes.h
+++ b/net/can/Can_GeneralTypes.h
@@ -41,19 +41,6 @@
 #define TPL_CAN_CLASSIC_FRAME_MAXIMUM_PAYLOAD_SIZE (8)
 #define TPL_CAN_FD_FRAME_MAXIMUM_PAYLOAD_SIZE (64)
 
-/**
- * @typedef tpl_can_protocol_version_t
- *
- * Select the version of the CAN protocol to operate for a specific CAN
- * controller.
- */
-typedef enum
-{
-	CAN_PROTOCOL_VERSION_CLASSIC,
-	CAN_PROTOCOL_VERSION_FD,
-	CAN_PROTOCOL_VERSION_XL
-} tpl_can_protocol_version_t;
-
 /**
  * @typedef Can_IdType
  *
@@ -149,7 +136,6 @@ typedef struct tpl_can_controller_t tpl_can_controller_t;
 struct tpl_can_controller_config_t
 {
 	tpl_can_controller_t *controller;
-	tpl_can_protocol_version_t protocol_version;
 	CanControllerBaudrateConfig baud_rate_config;
 };
 typedef struct tpl_can_controller_config_t tpl_can_controller_config_t;
-- 
2.34.1

